include/Mathematical/SquareMatrix.hpp

```diff
@@ -311,9 +327,81 @@ class SquareMatrix : public MatrixX<_Scalar> {
 
         return value;
     }
-};
 
 
+    /**
+     *  @return an non-pivoted LU decomposition in an array, with L at position 0 and U on position 1 of the array.
+     *   warning: Pivoting is required to ensure that the decomposition is stable. Eigen3 provides partial and full pivot modules.
+     */
+    std::array<Self, 2> NoPivotLUDecomposition() const {
+
+        auto M = this->get_dim();
+
+        Self L = Self::Zero(M, M);
+        Self U = Self::Zero(M, M);
+        
+        // Algorithm from "https://www.geeksforgeeks.org/doolittle-algorithm-lu-decomposition/"
+        for (int i = 0; i < M; i++) { 
+    
+            // Upper Triangular 
+            for (int k = i; k < M; k++) { 
+    
+                // Summation of L(i, j) * U(j, k) 
+                int sum = 0; 
+                for (int j = 0; j < i; j++) 
+                    sum += (L(i,j) * U(j,k)); 
+    
+                // Evaluating U(i, k) 
+                U(i,k) = this->operator()(i,k) - sum; 
+            } 
+    
+            // Lower Triangular 
+            for (int k = i; k < M; k++) { 
+                if (i == k) 
+                    L(i,i) = 1; // Diagonal as 1 
+                else { 
+    
+                    // Summation of L(k, j) * U(j, i) 
+                    int sum = 0; 
+                    for (int j = 0; j < i; j++) 
+                        sum += (L(k,j) * U(j,i)); 
+    
+                    // Evaluating L(k, i) 
+                    L(k,i) = (this->operator()(k,i) - sum) / U(i,i); 
+                } 
+            } 
+        } 
+        
+        // Test if the decomposition was stable
+        Self A = Self(L * U);
```

Comment:
```suggestion
        Self A = L * U;
```

I don't think it's necessary to cast `L * U` into `Self`.